<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>910</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QIno - UI Interface for ino arduino tool</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resource.qrc">
    <normaloff>:/icons/images/arduino.png</normaloff>:/icons/images/arduino.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>-1</number>
      </property>
      <property name="documentMode">
       <bool>true</bool>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>910</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_New"/>
    <addaction name="action_Save"/>
    <addaction name="action_Close"/>
    <addaction name="action_Save_As"/>
    <addaction name="separator"/>
    <addaction name="action_Exit"/>
   </widget>
   <widget class="QMenu" name="menu_Project">
    <property name="title">
     <string>&amp;Project</string>
    </property>
    <widget class="QMenu" name="menuRecently_Opened">
     <property name="title">
      <string>Recently Opened</string>
     </property>
     <addaction name="separator"/>
    </widget>
    <addaction name="action_New_Project"/>
    <addaction name="action_Open_Project"/>
    <addaction name="menuRecently_Opened"/>
   </widget>
   <widget class="QMenu" name="menu_Build">
    <property name="title">
     <string>&amp;Build</string>
    </property>
    <addaction name="action_Build"/>
    <addaction name="action_Clean"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About"/>
   </widget>
   <widget class="QMenu" name="menu_Tools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="action_Serial_Monitor"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Project"/>
   <addaction name="menu_Build"/>
   <addaction name="menu_Tools"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="acceptDrops">
    <bool>true</bool>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_New"/>
   <addaction name="action_Save"/>
   <addaction name="action_Save_As"/>
   <addaction name="separator"/>
   <addaction name="action_New_Project"/>
   <addaction name="action_Open_Project"/>
   <addaction name="separator"/>
   <addaction name="action_Build"/>
   <addaction name="action_Clean"/>
   <addaction name="action_Upload"/>
   <addaction name="separator"/>
   <addaction name="action_Serial_Monitor"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="allowedAreas">
    <set>Qt::BottomDockWidgetArea|Qt::TopDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Console</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout">
     <item row="2" column="1">
      <widget class="QToolButton" name="cleanConsoleBtn">
       <property name="toolTip">
        <string>Clean Console</string>
       </property>
       <property name="text">
        <string>Clean</string>
       </property>
       <property name="icon">
        <iconset theme="edit-clear">
         <normaloff/>
        </iconset>
       </property>
      </widget>
     </item>
     <item row="0" column="0" rowspan="3">
      <widget class="QTextEdit" name="consoleOut">
       <property name="undoRedoEnabled">
        <bool>false</bool>
       </property>
       <property name="lineWrapMode">
        <enum>QTextEdit::NoWrap</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="acceptRichText">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QToolButton" name="stopBtn">
       <property name="toolTip">
        <string>Stop Process</string>
       </property>
       <property name="text">
        <string>Stop</string>
       </property>
       <property name="icon">
        <iconset theme="process-stop">
         <normaloff/>
        </iconset>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_2">
   <property name="windowTitle">
    <string>Project</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <widget class="ProjectTree" name="projectTree">
       <property name="contextMenuPolicy">
        <enum>Qt::ActionsContextMenu</enum>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::InternalMove</enum>
       </property>
       <property name="defaultDropAction">
        <enum>Qt::MoveAction</enum>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::ExtendedSelection</enum>
       </property>
       <property name="animated">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_Build">
   <property name="icon">
    <iconset theme="run-build">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Build</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="action_New">
   <property name="icon">
    <iconset theme="document-new">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_Save_As">
   <property name="icon">
    <iconset theme="document-save-as">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Save As...</string>
   </property>
  </action>
  <action name="action_Exit">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_Clean">
   <property name="icon">
    <iconset theme="run-build-clean">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Clean</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="action_Open_Project">
   <property name="icon">
    <iconset theme="project-open">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Open Project</string>
   </property>
  </action>
  <action name="action_Close">
   <property name="icon">
    <iconset theme="document-close">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="action_New_Project">
   <property name="icon">
    <iconset theme="project-development-new-template">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;New Project</string>
   </property>
  </action>
  <action name="action_Upload">
   <property name="icon">
    <iconset theme="arrow-up-double">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Upload</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="action_Serial_Monitor">
   <property name="icon">
    <iconset resource="../resource.qrc">
     <normaloff>:/icons/images/connection.png</normaloff>:/icons/images/connection.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Serial Monitor</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ProjectTree</class>
   <extends>QTreeView</extends>
   <header>projecttree.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_Exit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cleanConsoleBtn</sender>
   <signal>clicked()</signal>
   <receiver>consoleOut</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>889</x>
     <y>381</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>495</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
